codegen/winterfell/src/air/boundary_constraints.rs

```diff
@@ -56,20 +58,59 @@ fn add_assertions(func_body: &mut codegen::Function, ir: &Air, trace_segment: Tr
     // declare the result vector to be returned.
     func_body.line("let mut result = Vec::new();");
 
-    // add the boundary constraints
+    // add the main boundary constraints
     for constraint in ir.boundary_constraints(trace_segment) {
```

Comment:
I understand why the comment is practically correct here - even if `trace_segment == 1`, we will not enter the loop because there shouldn't be boundary constraints over auxiliary trace segment yet. But it sees like this `add_assertions()` function is really two separate functions combined. I wonder if a clear way would be to split it into two separate functions - one for the main trace segment and one for busses.